client/fs_endpoint.go

```diff
@@ -704,6 +703,11 @@ OUTER:
 			continue
 		}
 
+		// When eof and cancel received then cancel
```

Comment:
I took a crack at expanding this comment for clarity:

```suggestion
		// At this point we can stop without waiting for more changes,
		// because we have EOF and either we're not following at all,
		// or the channel returned by blockUntilNextLog has closed
```

Reply:
In the last line I would say that "eof was received and last read was executed", because it doesn't include `blockUntilNextLog` (it's an internal name from a different function) and the information that it's closed is incorrect (it only emitted)

Reply:
Updated